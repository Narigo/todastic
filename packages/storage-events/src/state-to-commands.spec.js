const stateToCommands = require("./state-to-commands.js");

describe("stateToCommands", () => {
  it("returns an empty list for empty state", () => {
    expect(stateToCommands([])).toEqual([]);
  });

  it("returns a single ADDED_TODO for a single todo", () => {
    // prettier-ignore
    const myTodos = [ { children: [], todoId: "5bce72e9986417accf87547e-id-1-1-2", labels: [], position: 0, status: "open", title: "Test title" } ];
    // prettier-ignore
    const expected = [ { type: "ADD_TODO", data: { children: [], labels: [], parentId: null, position: 0, status: "open", title: "Test title" } } ];
    expect(stateToCommands(myTodos)).toEqual(expected);
  });
});
