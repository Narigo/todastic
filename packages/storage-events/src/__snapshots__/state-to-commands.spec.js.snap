// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stateToCommands returns working complex ADD_TODO commands 1`] = `
Array [
  Object {
    "data": Object {
      "children": Array [
        Object {
          "children": Array [],
          "labels": Array [],
          "position": 0,
          "status": "open",
          "title": "One-One",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "labels": Array [],
              "position": 0,
              "status": "open",
              "title": "One-Two-One",
            },
            Object {
              "children": Array [],
              "labels": Array [],
              "position": 1,
              "status": "open",
              "title": "One-Two-Two",
            },
          ],
          "labels": Array [],
          "position": 1,
          "status": "open",
          "title": "One-Two",
        },
        Object {
          "children": Array [],
          "labels": Array [],
          "position": 2,
          "status": "open",
          "title": "One-Three",
        },
      ],
      "labels": Array [],
      "parentId": null,
      "position": 0,
      "status": "open",
      "title": "One",
    },
    "type": "ADD_TODO",
  },
  Object {
    "data": Object {
      "children": Array [
        Object {
          "children": Array [],
          "labels": Array [],
          "position": 0,
          "status": "open",
          "title": "Two-One",
        },
        Object {
          "children": Array [],
          "labels": Array [],
          "position": 1,
          "status": "open",
          "title": "Two-Two",
        },
      ],
      "labels": Array [],
      "parentId": null,
      "position": 1,
      "status": "open",
      "title": "Two",
    },
    "type": "ADD_TODO",
  },
  Object {
    "data": Object {
      "children": Array [],
      "labels": Array [],
      "parentId": null,
      "position": 2,
      "status": "open",
      "title": "Three",
    },
    "type": "ADD_TODO",
  },
  Object {
    "data": Object {
      "children": Array [
        Object {
          "children": Array [],
          "labels": Array [],
          "position": 0,
          "status": "open",
          "title": "Four-One",
        },
        Object {
          "children": Array [],
          "labels": Array [],
          "position": 1,
          "status": "open",
          "title": "Four-Two",
        },
      ],
      "labels": Array [],
      "parentId": null,
      "position": 3,
      "status": "open",
      "title": "Four",
    },
    "type": "ADD_TODO",
  },
]
`;
